### DISCUSSION TITLE
Problems migrating from moor to drift

### DISCUSSION THREAD

**Discussion by MiguelSOliveira** [simolus3/drift#2135] [Q&A] [answered]
hi!

I'm trying to migrate from moor 4.3 to drift 1, by following the migration guide.

however I'm having a lot of problems with generated_migrations/schema_vOld versions. 

Mainly this error:
```
../../../.pub-cache/hosted/pub.dartlang.org/drift-1.7.1/lib/src/runtime/query_builder/schema/entities.dart:81:30: Context: 'ResultSetImplementation.attachedDatabase' is
defined here.
  DatabaseConnectionUser get attachedDatabase;
```

I've already manually replaced GeneratedXColumn by GeneratedColumn<X> and it seemed to have worked but the above error has me stumped.

Thank you!

**Comment by simolus3** [comment, marked as answer]
Thanks for the report! It looks like you've posted the context of that error, but not the full error message from the compiler. Could you add that as well?

> I've already manually replaced GeneratedXColumn by GeneratedColumn

Ideally, you shouldn't have to do that. You can just run `dart run drift_dev schema generate ...` to make drift re-generate the test helper files with the latest state.

**Reply by MiguelSOliveira** [reply]
Oh thanks Simon, that was the problem. I needed to run the generate command. May I suggest maybe adding a quick note on the migration doc? It was the first document I saw with the migration and it doesn't mention this anywhere there, from what I can see?

Anyways, problem solved, ty!

**Reply by simolus3** [reply]
Good point, I've added a note about this in dfa01c133276a864f44a17f16921f34a51db3355.
